service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ColleaguesToNotifyController.java

```diff
@@ -0,0 +1,119 @@
+            return respond(caseDetails, errors);
+        }
+
+        // Use the first party we are representing - solicitors shouldn't be representing multiple
+        // todo - what about children's solicitors...
+        caseDetails.getData().put("respondentName", represented.get(0).toParty().getFullName());
+        caseDetails.getData().put("colleaguesToNotify",
+            represented.get(0).getSolicitor().getColleaguesToBeNotified());
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackrequest) {
+        final CaseData caseData = getCaseData(callbackrequest);
+        final CaseDetailsMap caseDetails = caseDetailsMap(callbackrequest.getCaseDetails());
+
+        List<SolicitorRole> roles = caseRoleLookupService.getCaseSolicitorRolesForCurrentUser(caseData.getId());
+        List<WithSolicitor> represented = roles.stream()
+            .map(role -> role.getRepresentedPerson(caseData))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .map(Element::getValue)
+            .collect(Collectors.toList());
+
+        List<String> errors;
+        if (represented.isEmpty()) {
+            errors = List.of("There is no one this user is representing on this case.");
+            return respond(caseDetails, errors);
+        }
+
+        if (roles.get(0).getRepresenting().equals(SolicitorRole.Representing.RESPONDENT)) {
+            // Update the respondent who's solicitor it is
+            List<Element<Respondent>> respondents = caseData.getRespondents1();
+
+            for (Element<Respondent> respondent : respondents) {
+                if (respondent.getValue().equals(represented.get(0))) {
+                    respondent.getValue().getSolicitor().setColleaguesToBeNotified(caseData.getColleaguesToNotify());
+                }
+            }
+
+            caseDetails.put("respondents1", respondents);
+        } else if (roles.get(0).getRepresenting().equals(SolicitorRole.Representing.CHILD)) {
+            List<Element<Child>> children = caseData.getChildren1();
+
+            for (Element<Child> child : children) {
+                // Update all children we represent
+                if (represented.contains(child.getValue())) {
+                    child.getValue().getSolicitor().setColleaguesToBeNotified(caseData.getColleaguesToNotify());
+                }
+            }
+
+            caseDetails.put("children1", children);
+        }
+
+        // Remove temporary fields as not needed anymore
+        caseDetails.removeAll("respondentName", "colleaguesToNotify");
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/submitted")
+    public void handleSubmitted(@RequestBody CallbackRequest callbackRequest) {
+        final CaseData caseData = getCaseData(callbackRequest);
+
+        // Send emails!
+    }
```

Comment:
Waiting for confirmation for implementing something?

Reply:
ah yes I think originally I was thinking we should be notifying people they were added as colleagues but doesn't look like thats a requirement so will remove this callback